c4spar/deno-cliffy
# fix(command): support zsh fpath completion

This potentially closes #702.

These changes are based on patterns seen in other tools that generate zsh completions. If you run `deno completions zsh` you should be able to compare.

- ensure `#compdef <cmd>` is on the first line
- DO NOT add the `(( $+functions[<cmd>] ))` check on the root command. Doing so errors about function nesting. If you look, the `_deno` completions omit this on the root `_deno` function.
- The end block checks `funcstack` and runs the `_<cmd> $@` or falls back to the `compgen command`. This was copied from the deno completions and I believe this will work with both sourcing the completions directory or being used on the `fpath`. That said, I've never seen `funcstack` before so I'm not sure 100%.

Steps to test:

1 .Run `deno compile -A --output examples/cliffy examples/commands/demo.ts`
2. Run `examples/cliffy completions zsh > /path/to/zsh/site-functions/_cliffy`
3. Make sure you have `fpath=(/path/to/zsh/site-functions)` set and `compinit` called in your zsh setup
4. Reload shell
5. `examples/cliffy` + `<tab>` will auto complete now
